paasta_tools/kubernetes_tools.py

```diff
@@ -2330,13 +2405,14 @@ def get_node_affinity(self) -> Optional[V1NodeAffinity]:
                 for key, op, vs in requirements
             ]
         )
-        selector = V1NodeSelector(node_selector_terms=[term])
+
         return V1NodeAffinity(
-            # this means that the selectors are only used during scheduling.
-            # changing it while the pod is running will not cause an eviction.
-            # this should be fine since if there are whitelist/blacklist config
-            # changes, we will bounce anyway.
-            required_during_scheduling_ignored_during_execution=selector,
+            required_during_scheduling_ignored_during_execution=V1NodeSelector(
+                node_selector_terms=[required_term]
+            ),
+            preferred_during_scheduling_ignored_during_execution=preferred_terms
+            if preferred_terms
+            else None,
```

Comment:
might be nice to do something like 
```python
preferred_terms = preferred_terms or None
return V1NodeAffinity(...)
```
to avoid the formatting here, but that's not a big deal